### PR TITLE
to-do list CRUD

### PR DESCRIPTION
**Pull request by bronnie313**
## CRUD for the to-do list
- Created elements so that the to-do list project is functional
- created a function for adding a new task to the array and then stored in local storage
- created a function for editing the task and it is saved by pressing enter
- made the delete button functional and when the user edits the task, the changes are stored in the local storage directly

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,39 +1,77 @@
...
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  addtask(task.value);
+  displayTask();
+});
+
+// deleting task
+const deleteTask = (index) => {
+  listName.splice(index, 1);
+  localStorage.setItem('tasks', JSON.stringify(listName));
```
Comment (Shedrack-Sunday) [inline src/index.js:66]: - [ ] Based on project requirements, all new functionality should be implemented in a new file. Kindly create a new javascript file, and do so based on the requirements below 👍 

## Instructions:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hi @bronnie313      👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well implementing the requirements,  However, there are a few you need to work on, to move to the next project. 💯

 



Highlights
✅All linters are checked
✅ Proper PR summary 
  
# Required Changes ♻️

- [x] I must admit you have really put in the effort so far in this project, however, your weback did not run successfully. kindly fix the error as one o them has to do with the development `mode` not being set

## Error:
image

_check the comment section for a detailed review_ 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/index.js (hunk 1 above)
Comment (bronnie313) [reply src/index.js:66]: thats a new file all functions are implemented i deleted the first one i used

File: src/index.js [hunk 2]
```diff
@@ -1,39 +1,77 @@
...
   }
+  localStorage.setItem('tasks', JSON.stringify(listName));
+  displayTask();
 };
 
-populate();
+window.onload = () => {
+  displayTask();
+  deleteTask();
+};
```
Comment (AlphaNtihinduka) [inline src/index.js:77]: - [ ] You have so far done well implementing most of the functionalities of the to-do list. However, let's make sure we work on the delete function as intended. If you are keen, when the window is reloaded, the every task at array[0] is deleted. In addition, the delete button is not deleting the tasks intended to delete. Kindly fix that to make sure that our to-do list is professional ☺️ 

Video presentation 😎 


https://user-images.githubusercontent.com/99394374/223659818-1f00d4e7-fb65-4bc1-93ff-11b324dffd93.mp4

**Review by AlphaNtihinduka — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 
## Hi @bronnie313 🙋‍♂️, 

Your project is complete!  There is nothing else to say other than... Great job 👏 

approved-accepted

Congratulations! 🎉 
TO HIGHLIGHT:

- Your project is professional :heavy_check_mark: 
- Your repo is professional :heavy_check_mark: 
- Well-structured files :heavy_check_mark: 
- Professional README.md file 💯
- No linter errors :heavy_check_mark: 

Now you can merge this branch and move on :1st_place_medal: 


## Optional suggestions
- Kindly refer to the inline optional suggestions made. Ensure to apply them to have a professional project that suits recruiters' needs.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HAPPY CODING AND GOOD LUCK WITH THE COMING ONES :crossed_fingers: :four_leaf_clover:

**Pull request merged**